### PR TITLE
NS-1013: Update useLocalStorage hook

### PR DESCRIPTION
**Pull request by vyrwu**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,20 +9,33 @@
   "repository": "git@github.com:bence-toth/react-hook-local-web-storage.git",
   "license": "LGPL",
   "author": "Bence A. Toth <tothab@gmail.com>",
+  "contributors": [
+    "Aleksander Nowak (https://github.com/vyrwu)"
+  ],
   "main": "dist/index.js",
+  "dependencies": {
```
Comment (bence-toth) [inline package.json]: Not sure why we need these dependencies

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,26 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (bence-toth) [inline .gitignore]: Not sure why we need all this on the `.gitignore` file

File: src/index.js [hunk 3]
```diff
@@ -37,7 +37,31 @@ const useLocalStorage = (key, {updateFrequency}) => {
...
-export default useLocalStorage
+const useLocalStorageNoSync = key => {
+  const [value, setValue] = useState()
+
+  const read = () => {
+    const oldValue = value
+    const newValue = localStorage.getItem(key)
+    if (newValue !== oldValue) {
+      setValue(newValue)
+    }
```
Comment (bence-toth) [inline src/index.js]: Shouldn't this return `newValue`?

File: src/index.js [hunk 4]
```diff
@@ -37,7 +37,31 @@ const useLocalStorage = (key, {updateFrequency}) => {
...
+    else {
+      localStorage.removeItem(key)
+    }
+    setValue(newValue)
+  }
+
+  return [read, write]
+}
+
+export default {useLocalStorage, useLocalStorageNoSync}
```
Comment (bence-toth) [inline src/index.js]: Missing new line EOF

File: package.json (hunk 1 above)
Comment (vyrwu) [reply package.json]: changed Babel deps to --dev-only in 5e75fee394bee600d0207f1ec14506d8ff7b4b39

Babel output only adds semicolons to the code. Maybe ditch?

File: src/index.js (hunk 3 above)
Comment (vyrwu) [reply src/index.js]: Returns `newValue` in 5e75fee394bee600d0207f1ec14506d8ff7b4b39

File: .gitignore (hunk 2 above)
Comment (vyrwu) [reply .gitignore]: Cleaden-up in 5e75fee394bee600d0207f1ec14506d8ff7b4b39

File: src/index.js (hunk 4 above)
Comment (vyrwu) [reply src/index.js]: Added in 20f4b58108e927ae0b84166d13eca28bf6149ce5

File: package.json (hunk 1 above)
Comment (vyrwu) [reply package.json]: Removed babel in ec7f4efd92d7e3cfeeba05e9cd0e8a4ea21b7622

File: package.json [hunk 5]
```diff
@@ -1,29 +1,33 @@
...
+  ],
+  "dependencies": {
     "react": "^16.8.6"
   },
-  "peerDependencies": {
-    "react": "^16.8.0"
+  "eslintConfig": {
+    "extends": "react-app"
+  },
+  "browserslist": {
```
Comment (bence-toth) [inline package.json]: This is for Babel, which is not used.

File: .gitignore [hunk 6]
```diff
@@ -1,2 +1,5 @@
+# dependencies
 /node_modules
-/dist
\ No newline at end of file
+
+# Babel output
```
Comment (bence-toth) [inline .gitignore]: This is not very accurate, as Babel is gone from the deps. Can you just revert all changes this file please?

File: src/index.js [hunk 7]
```diff
@@ -3,15 +3,15 @@ import {useState, useEffect} from 'react'
 const useLocalStorage = (key, {updateFrequency}) => {
   const [value, setValue] = useState()
 
-  const readFromLocalStorage = () => {
+  const read = () => {
```
Comment (bence-toth) [inline src/index.js]: Why changes these names (same for `write`)?

File: package.json (hunk 5 above)
Comment (vyrwu) [reply package.json]: Removed in 22b43cf8b3594215f9d53d883466d45c7dceff56

File: .gitignore (hunk 6 above)
Comment (vyrwu) [reply .gitignore]: Reverted in 22b43cf8b3594215f9d53d883466d45c7dceff56

File: src/index.js (hunk 7 above)
Comment (vyrwu) [reply src/index.js]: Just for readability. But I reverted these changes in 22b43cf8b3594215f9d53d883466d45c7dceff56

File: README.md [hunk 8]
```diff
@@ -51,6 +51,8 @@ const ComponentWithLocalStorage = () => {
 }
 ```
 
+**NOTE:** you can use `useLocalStorageNoSync` hook to opt out of using updates. First entry of the output array will then contain a function returning value stored in `localStorage` on demand.
```
Comment (bence-toth) [inline README.md]: Let's turn this note into a fully featured section the following way:

- Let's turn _Tweaking update frequency_ headling level 3 and move it under _Basic usage_.

- Let's rename this section to _Tweaking **the** update frequency_.

- Split _Caveats_ into a _Caveats_ (level 3) subsection under _Basic usage_ describing the possible performance issue if the update frequency is low, and another level 2 section called _Limitations_ where the _Caveats_ section currently is, and keep the note about the type limitations there.

- Make a new section called _Disable syncing_ for the new hook, elaborate a little bit on its usage, and show a working example similar to the one for the other hook.

File: .gitignore [hunk 9]
```diff
@@ -1,2 +1 @@
 /node_modules
-/dist
```
Comment (bence-toth) [inline .gitignore]: We need to keep `/dist` on `.gitignore`

File: package.json [hunk 10]
```diff
@@ -1,29 +1,18 @@
...
-    "build:watch": "babel src --out-dir dist --watch"
-  },
-  "devDependencies": {
-    "@babel/cli": "^7.5.0",
-    "@babel/core": "^7.5.0",
-    "@babel/preset-env": "^7.5.0",
-    "@babel/preset-react": "^7.0.0",
-    "babel-core": "^7.0.0-bridge",
+  "contributors": [
+    "Aleksander Nowak (https://github.com/vyrwu)"
```
Comment (bence-toth) [inline package.json]: Should follow this format: https://docs.npmjs.com/files/package.json#people-fields-author-contributors

File: src/index.js [hunk 11]
```diff
@@ -40,4 +40,32 @@ const useLocalStorage = (key, {updateFrequency}) => {
...
+    else {
+      localStorage.removeItem(key)
+    }
+    setValue(newValue)
+  }
+
+  return [readFromLocalStorage, writeToLocalStorage]
+}
+
+export default {
```
[the remark this answers is not in the document]
Comment (vyrwu) [reply src/index.js]: Changed to named import in e8e98e2065fc0310fbe1fb02ac454c8199f5f7f2

File: README.md (hunk 8 above)
Comment (vyrwu) [reply README.md]: Included in e8e98e2065fc0310fbe1fb02ac454c8199f5f7f2

File: .gitignore (hunk 9 above)
Comment (vyrwu) [reply .gitignore]: Included in e8e98e2065fc0310fbe1fb02ac454c8199f5f7f2

File: package.json (hunk 10 above)
Comment (vyrwu) [reply package.json]: Updated in e8e98e2065fc0310fbe1fb02ac454c8199f5f7f2

**Pull request merged**